### PR TITLE
[IMP] mail, *: move code from components to models (ThreadIcon)

### PR DESCRIPTION
**Pull request by oomsveta**
Task-3002515.

\* = hr_holidays, im_livechat

### CODE REVIEW

File: addons/mail/static/src/components/chat_window_header/chat_window_header.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
                 <t t-if="chatWindow.thread and chatWindow.thread.model === 'mail.channel'">
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window_header/chat_window_header.xml:12]: condition can simply become `chatWindowHeaderView.threadIconView`

**Review by alexkuhn — APPROVED** [review]
LGTM otherwise

**Comment by alexkuhn** [comment]
Also re-target `master` instead of `16.0`

**Comment by oomsveta** [comment]
@alexkuhn Changes done, commits squashed, and PR re-targeted 👌

File: addons/mail/static/src/components/chat_window_header/chat_window_header.xml [hunk 2]
```diff
@@ -9,10 +9,10 @@
                         <i class="fa fa-arrow-left"/>
                     </div>
                 </t>
-                <t t-if="chatWindow.thread and chatWindow.thread.model === 'mail.channel'">
+                <t t-if="chatWindowHeaderView.threadIconView'">
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window_header/chat_window_header.xml:12]: typo: remove `'` (antepenultimate character on this line, without considering EOL)

**Comment by alexkuhn** [comment]
Better make another branch with prefix `16.0-`. Current branch is being mistakenly considered as a `16.0`, which makes `ci/upgrade` fail

**Comment by oomsveta** [comment]
> Better make another branch with prefix `16.0-`. Current branch is being mistakenly considered as a `16.0`, which makes `ci/upgrade` fail

I want to try one last thing if you don't mind; let's see if it can fix the failing CI check

**Review by alexkuhn — CHANGES_REQUESTED** [review]
There's some old and ugly code that show the template of thread icon as a part of a notification to display (whether toaster or native notification):
https://github.com/odoo/odoo/blob/master/addons/mail/static/src/models/messaging_notification_handler.js#L657

I think it's best to just remove the `ThreadIcon` part here.
We might reintroduce this small feature later when notifications are properly modeled (as this is a necessity for `ThreadIconView`)

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**